addons/sale_stock/tests/test_sale_stock.py

```diff
@@ -734,3 +734,64 @@ def test_12_return_without_refund(self):
         return_picking.button_validate()
         # Checks the delivery amount (must still be 10).
         self.assertEqual(sale_order.order_line.qty_delivered, 10)
+
+    def test_08_sale_return_qty_and_cancel(self):
+        """
+        Test a SO with a product on delivery. Deliver the SO,
+        then do a return of the delivered picking (for all quantities).
+        Then cancel Sale order, it won't raise any warning, it should be cancelled.
+        """
```

Comment:
some modifications to the test data:
- use `self.partner` as customer (from the TestSale class) - will be compatible with `master-demo-data-yti` (#37312) out of the box as it's about to be merged
- use `self.products['prod_del']` for the deliverable 'or another product that is created in a setup of this test class)